92-AppendixC.Rmd

```diff
@@ -0,0 +1,312 @@
+
+In the example below, we use {readr} to load a CSV file named 'anes_timeseries_2020_csv_20220210.csv' into an R object called `anes_csv`. The `read_csv()` imports the file and stores the data in the `anes_csv` object. We can then use this object for further analysis.
+
+```r
+library(readr)
+
+anes_csv <-
+  read_csv("data/anes_timeseries_2020_csv_20220210.csv")
+```
+
+## Loading Excel files into R
+
+Excel, a widely used spreadsheet software program created by Microsoft, is a common file format in survey research. We can load Excel spreadsheets into the R environment using the {readxl} package. The package supports both the legacy `.xls` files and the modern `.xlsx` format. 
+
+To load Excel data into R, we can use the `read_excel()` function from the {readxl} package. This function offers a range of customizable options for the import process. Let's explore the syntax:
+
+```
+read_excel(
+  path,
+  sheet = NULL,
+  range = NULL,
+  col_names = TRUE,
+  col_types = NULL,
+  na = "",
+  trim_ws = TRUE,
+  skip = 0,
+  n_max = Inf,
+  guess_max = min(1000, n_max),
+  progress = readxl_progress(),
+  .name_repair = "unique"
+)
+```
+
+The arguments are:
+
+* `path`: the path to the Excel file to import
+* `sheet`: the name or index of the sheet (sometimes called tabs) within the Excel file
+* `range`: the range of cells to import (for example, "P15:T87")
+* `col_names`: indicates whether the first row of the dataset contains column names
+* `col_types`: specify the data types of columns
+* `na`: define the representation of missing values (for example, "NULL")
+* `trim_ws`: controls whether leading and trailing whitespaces should be trimmed
+* `skip` and `n_max`: enable skipping rows and limit the number of rows imported
+* `guess_max`: sets the maximum number of rows used for data type guessing
+* `progress`: specifies a progress bar for large imports
+* `.name_repair`: Determines how column names are repaired if they are not valid
+
+In the code example below, we load an Excel spreadsheet named 'anes_timeseries_2020_csv_20220210.xlsx' into R. The resulting data is saved as a tibble in the `anes_excel` object, ready for further analysis.
+
+```r
+library(readxl)
+
+anes_excel <-
+  read_excel(path = "data/anes_timeseries_2020_csv_20220210.xlsx")
+```
+
+## Importing Stata, SAS, and SPSS files into R
+
+The {haven} package, also from the tidyverse ecosystem, imports various proprietary data formats: Stata `.dta` files, SAS `.sas7bdat` and `.sas7bcat` files, and SPSS `.sav` files. One of the notable strengths of the {haven} package is its ability to handle multiple proprietary formats within a unified framework. It offers dedicated functions for each supported proprietary format, making it straightforward to import data. Here, we introduce `read_dat()` for Stata files, `read_sas()` for SAS files, and `read_sav()` for SPSS files.
+
+### Syntax
+
+Let's explore the syntax for importing Stata files `.dat` files using `haven::read_dat()`:
+
+```r
```

Comment:
you provide this for stata, should we also do this for sas and spss?  we should be consistent.

Reply:
Added!